dataframe/dataframe.go

```diff
@@ -434,6 +434,62 @@ func (df DataFrame) GroupBy(colnames ...string) *Groups {
 	return groups
 }
 
+type ColumnValuesPerSubset struct {
+	Values      series.Series
+	SubsetIndex series.Indexes
+}
+
+type ColumnUpdateRule struct {
+	ColName      string
```

Comment:
Can we be consistent on naming, `ColName` should be `ColumnName`. Or if we cal it `ColName` then the other field should be `ColValues`.